### PR TITLE
Jo 13289

### PR DESCRIPTION
**Pull request by jeffrey-opdam**
Thank you for your pull request!
By completing this pull request, you agree to the [Contributing License Agreement](https://github.com/ALM-Rangers/Team-Services-Bot/blob/master/.github/CLA.md).

Fixes # .

Changes proposed in this pull request:  
- Added unit tests.
- Simplified them.
- Start using Fluent Assertions.
- 

@ALM-Rangers/Team-Services-Bot

### CODE REVIEW

File: src/Common/TestCodeRules.ruleset [hunk 1]
```diff
@@ -1,9 +1,12 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
-<RuleSet Name="Test code ruleset" Description="This rule set contains all rules except a few which are not appropriate for test code." ToolsVersion="14.0">
+<RuleSet Name="Test code ruleset" Description="This rule set contains all rules except a few which are not appropriate for test code." ToolsVersion="15.0">
   <IncludeAll Action="Warning" />
   <Rules AnalyzerId="Microsoft.Analyzers.ManagedCodeAnalysis" RuleNamespace="Microsoft.Rules.Managed">
     <Rule Id="CA1014" Action="None" />
     <Rule Id="CA1020" Action="None" />
+    <Rule Id="CA1506" Action="None" />
```
Comment (jeffrey-opdam) [inline src/Common/TestCodeRules.ruleset]: Bootstrapping for tests, so a lot of coupling is expected.

File: src/Common/TestCodeRules.ruleset [hunk 2]
```diff
@@ -1,9 +1,12 @@
...
+    <Rule Id="CA1707" Action="None" />
```
Comment (jeffrey-opdam) [inline src/Common/TestCodeRules.ruleset]: Allowing underscores in tests, to make the method(s) and what is shown in the tests results more readable.

Comment (Hussar77) [inline src/Common/TestCodeRules.ruleset]: I personally dislike underscores in tests names. We expected to have tons of the tests and you'll look into the name only when test willl fail. For me test list is readable enough without underscores.

**Comment by jeffrey-opdam** [comment]
No. Whenever a wait is called, the framework will wait for an Incoming activity / dialog. So the userdata that comes into the startasync differs from that comes into the private methods that are 'waited'.

**Pull request merged**